crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1043,22 +1025,23 @@ impl<'src> Parser<'src> {
 
     /// Parses a boolean operation expression.
     ///
-    /// Note that the boolean `not` operator is parsed as a unary operator and
-    /// not as a boolean operation.
+    /// Note that the boolean `not` operator is parsed as a unary expression and
+    /// not as a boolean expression.
     ///
     /// # Panics
     ///
     /// If the parser isn't positioned at a `or` or `and` token.
     ///
     /// See: <https://docs.python.org/3/reference/expressions.html#boolean-operations>
-    fn parse_bool_operation_expression(
+    fn parse_boolean_expression(
         &mut self,
         lhs: Expr,
         start: TextSize,
-        operator_token: TokenKind,
-        operator_binding_power: OperatorPrecedence,
+        op: BoolOp,
         context: ExpressionContext,
     ) -> ast::ExprBoolOp {
+        self.bump(TokenKind::from(op));
```

Comment:
Nit: I would probably add a `kind` and `precedence` method to `BoolOp`. It improves readability and we don't realy need the `From` and `Into` infrastructure. 

Oh, or is this not possible (at least for precedence) because `OperatorPrecedence` is a parser and not an `AST` type.

Reply:
Yeah, `OperatorPrecedence` is part of the parser crate. Also, the AST crate cannot depend on the parser crate as it'll create a cyclic dependency, so we can't define the `kind` on `BoolOp`.